Source/Scene/Cesium3DTileset.js

```diff
@@ -68,10 +74,20 @@ define([
         }
         //>>includeEnd('debug');
 
-        url = appendForwardSlash(url);
+        var fileName;
+        if (getExtensionFromUri(url) === 'json') {
+            fileName = getFilenameFromUri(url);
+        } else {
+            fileName = 'tiles.json';
+        }
+
+        if ((url.lastIndexOf('/') !== (url.length - 1)) && (getExtensionFromUri(url) === '')) {
+            url = appendForwardSlash(url);
+        }
+        url = getBaseUri(url);
 
         this._url = url;
-        this._tilesJson = url + 'tiles.json';
+        this._tilesJson = url + fileName;
```

Comment:
I think this code can be cleaner. A structure like this might be better:

```javascript
var url;
var tilesJson;
if (getExtensionFromUri(url) === 'json') {
    // Set url and tilesJson
} else {
    // Set url and tilesJson
}
this._url = url;
this._tilesJson = tilesJson;
```